File: sdk-python/wappflow/resources/templates.py
from __future__ import annotations

from .._client import HttpClient
from .._types import Template


class TemplatesResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def list(self) -> list[Template]:
        """List all WhatsApp templates (approved and pending).

        Requires scope ``templates:read``.
        """
        result = self._http.get("/templates")
        return result.get("templates", [])

File: sdk/src/resources/templates.ts
import type { HttpClient } from "../http";
import type { Template } from "../types";

export class TemplatesResource {
  constructor(private http: HttpClient) {}

  /**
   * List all WhatsApp templates (approved and pending review).
   * Requires scope `templates:read`.
   */
  async list(): Promise<{ templates: Template[] }> {
    return this.http.get<{ templates: Template[] }>("/templates");
  }
}
